src/designmatrix.jl

```diff
@@ -55,38 +55,36 @@ function combineDesignmatrices(X1::DesignMatrix, X2::DesignMatrix)
         else
             fcomb = Vector{FormulaTerm}(undef, length(X1.formulas) + 1) # mass univariate case
         end
-        fcomb[1:end-1] = X1.formulas
-        fcomb[end] = X2.formulas
-        return DesignMatrix(fcomb, Xcomb, [X1.events..., X2.events])
+        fcomb[1:end-1] = formulas(X1)
+        fcomb[end] = formulas(X2)[1]
+        @debug typeof(Xcomb)
+        return T(fcomb, Xcomb, [events(X1)..., events(X2)])
     end
 end
 
 
+Base.:+(X1::Vector{T}, X2::T) where {T<:AbstractDesignMatrix} = [X1..., X2]
+Base.:+(X1::T, X2::T) where {T<:AbstractDesignMatrix} = [X1, X2]
+Base.:+(X1::Nothing, X2::AbstractDesignMatrix) = [X2]
 
-Base.:+(X1::DesignMatrix, X2::DesignMatrix) = combineDesignmatrices(X1, X2)
 
-Base.:+(X1::Nothing, X2::DesignMatrix) = X2
+# helper to get the fixef of lmm but the normal matrix elsewhere
+get_modelmatrix(modelmatrix::Tuple) = modelmatrix[1]
```

Comment:
find better name reflecting that with
get_modelmatrix => returns Vector(modelmatrices)
modelmatrix => returns combined modelmatrix (or vector of modelmatrix)